File: types/src/watermixer.ts
import { Topic } from '.';

export namespace Watermixer {
  export interface Data {
    finishMixTimestamp: number;
  }

  export const SAMPLE: Data = {
    finishMixTimestamp: 0,
  };

  export const getStartMixingTopic = (uid: string): Topic => ({
    request: `${uid}/event/relay1/request`,
    response: `${uid}/event/relay1/response`,
  });
}

File: app_mobile/lib/models/devices/watermixer.dart
import 'package:flutter/material.dart';

import '../device.dart';

class WatermixerData {
  int finishMixTimestamp;

  factory WatermixerData.fromJson(Map<String, dynamic> json) {
    return WatermixerData(
      finishMixTimestamp: json["finishMixTimestamp"],
    );
  }
  Map<String, dynamic> toJson() {
    return {
      'finishMixTimestamp': finishMixTimestamp,
    };
  }

  static DeviceTopic getStartMixingTopic(String uid) {
    return new DeviceTopic(
        request: '$uid/event/relay1/request',
        response: '$uid/event/relay1/response');
  }

  WatermixerData({@required this.finishMixTimestamp});
}
